cloud/aws/deploy/website.go

```diff
@@ -0,0 +1,450 @@
+	defaultCacheBehavior := &cloudfront.DistributionDefaultCacheBehaviorArgs{
+		TargetOriginId:       pulumi.String("publicOrigin"),
+		ViewerProtocolPolicy: pulumi.String("redirect-to-https"),
+		AllowedMethods: pulumi.StringArray{
+			pulumi.String("GET"),
+			pulumi.String("HEAD"),
+			pulumi.String("OPTIONS"),
+		},
+		CachedMethods: pulumi.StringArray{
+			pulumi.String("GET"),
+			pulumi.String("HEAD"),
+			pulumi.String("OPTIONS"),
+		},
+		ForwardedValues: &cloudfront.DistributionDefaultCacheBehaviorForwardedValuesArgs{
+			QueryString: pulumi.Bool(false),
+			Cookies: &cloudfront.DistributionDefaultCacheBehaviorForwardedValuesCookiesArgs{
+				Forward: pulumi.String("none"),
+			},
+		},
+		// could be added to stack config in the future
+		MinTtl:     pulumi.Int(0),
+		DefaultTtl: pulumi.Int(3600),
+		MaxTtl:     pulumi.Int(86400),
+		FunctionAssociations: cloudfront.DistributionDefaultCacheBehaviorFunctionAssociationArray{
+			&cloudfront.DistributionDefaultCacheBehaviorFunctionAssociationArgs{
+				EventType:   pulumi.String("viewer-request"),
+				FunctionArn: rewriteFun.Arn,
+			},
+		},
+	}
+
+	// For each API forward to the appropriate API gateway
+	for name, api := range a.Apis {
+		apiDomainName := api.ApiEndpoint.ApplyT(func(endpoint string) string {
+			return strings.Replace(endpoint, "https://", "", 1)
+		}).(pulumi.StringOutput)
+
+		origins = append(origins, &cloudfront.DistributionOriginArgs{
+			DomainName: apiDomainName,
+			OriginId:   pulumi.String(name),
+			CustomOriginConfig: &cloudfront.DistributionOriginCustomOriginConfigArgs{
+				OriginReadTimeout:    pulumi.Int(30),
+				OriginProtocolPolicy: pulumi.String("https-only"),
+				OriginSslProtocols: pulumi.StringArray{
+					pulumi.String("TLSv1.2"),
+					pulumi.String("SSLv3"),
+				},
+				HttpPort:  pulumi.Int(80),
+				HttpsPort: pulumi.Int(443),
+			},
+		})
+
+		orderedCacheBehaviors = append(orderedCacheBehaviors,
+			&cloudfront.DistributionOrderedCacheBehaviorArgs{
+				PathPattern: pulumi.Sprintf("api/%s/*", name),
+				// rewrite the URL to the nitric service
+				FunctionAssociations: cloudfront.DistributionOrderedCacheBehaviorFunctionAssociationArray{
+					&cloudfront.DistributionOrderedCacheBehaviorFunctionAssociationArgs{
+						EventType:   pulumi.String("viewer-request"),
+						FunctionArn: apiRewriteFun.Arn,
+					},
+				},
+				AllowedMethods: pulumi.ToStringArray([]string{"GET", "HEAD", "OPTIONS", "PUT", "POST", "PATCH", "DELETE"}),
+				CachedMethods:  pulumi.ToStringArray([]string{"GET", "HEAD", "OPTIONS"}),
+				TargetOriginId: pulumi.String(name),
+				ForwardedValues: &cloudfront.DistributionOrderedCacheBehaviorForwardedValuesArgs{
+					QueryString: pulumi.Bool(true),
+					Cookies: &cloudfront.DistributionOrderedCacheBehaviorForwardedValuesCookiesArgs{
+						Forward: pulumi.String("all"),
+					},
+					// Headers: pulumi.ToStringArray([]string{"*"}),
+				},
+				ViewerProtocolPolicy: pulumi.String("https-only"),
+			},
+		)
+	}
```

Comment:
same question as Azure on determinism when looping over map keys, could this produce changes to the distribution when none are necessary?

Reply:
For map loop determinism since go 1.23
```go
import (
	"maps"
	"slices"
)
...
for _, k := range slices.Sorted(maps.Keys(m)) {
    v := m[k]
}
```

Reply:
changes made